pkg/scheduler/api/podgroup_info/subgroup_info/subgroupset.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2025 NVIDIA CORPORATION
+// SPDX-License-Identifier: Apache-2.0
+
+package subgroup_info
+
+import (
+	"github.com/NVIDIA/KAI-scheduler/pkg/scheduler/api/topology_info"
+	"github.com/NVIDIA/KAI-scheduler/pkg/scheduler/log"
+)
+
+type SubGroupSet struct {
+	SubGroupInfo
+
+	groups  []*SubGroupSet
```

Comment:
Its a bit confusing the a SubGroupSet is a single subgroup.
The word set is not that good, because you have in here array of "sets" it gives the hint its a 2d array